pkg/controller/storagecluster/uninstall_reconciler_test.go

```diff
@@ -212,3 +212,47 @@ func assertGetStorageClusterEligibleNodes(
 	assert.NotEqual(t, 0, len(nodes.Items))
 	return nodes
 }
+
+func TestDeleteNodeTaint(t *testing.T) {
+	for _, eachPlatform := range allPlatforms {
+		cp := &CloudPlatform{platform: eachPlatform}
+		_, reconciler, sc, _ := initStorageClusterResourceCreateUpdateTestWithPlatform(t, cp, nil)
+
+		// case1: verify deleteNodeTaint deletes the default node taint
+
+		// Add default node taint on nodes
+		nodes := assertGetStorageClusterEligibleNodes(t, reconciler, sc)
+		for _, node := range nodes.Items {
+			new := node.DeepCopy()
+			new.Spec.Taints = append(new.Spec.Taints, corev1.Taint{Key: defaults.NodeTolerationKey, Effect: corev1.TaintEffectNoSchedule})
+
+			oldJSON, err := json.Marshal(node)
+			assert.NoError(t, err)
+
+			newJSON, err := json.Marshal(new)
+			assert.NoError(t, err)
+
+			patch, err := strategicpatch.CreateTwoWayMergePatch(oldJSON, newJSON, node)
+			assert.NoError(t, err)
+
+			err = reconciler.client.Patch(context.TODO(), &node, client.RawPatch(types.StrategicMergePatchType, patch))
+			assert.NoError(t, err)
+		}
```

Comment:
Can we do this processing in a separate function? It'd enhance readability of actual test cases.

Reply:
Done